sources/api/netdog/src/net_config.rs

```diff
@@ -0,0 +1,40 @@
+//! The net_config module contains the strucures needed to deserialize a `net.toml` file.
+//!
+//! These structures are the user-facing options for configuring 1 or more network interfaces.
+use serde::Deserialize;
+use std::collections::HashMap;
+
+// TODO: support deserializing different versions of this configuration.
+// Idea: write a deserializer that uses the `version` field and deserializes the rest of the config
+// into an enum with variants for each version, i.e.
+// enum NetConfig {
+//     V1(NetInterfaceV1)
+//     V2(NetInterfaceV2)
+// }
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetConfig {
+    // This attribute is needed to support versions, but we aren't using it yet
+    #[allow(dead_code)]
+    pub(crate) version: u8,
```

Comment:
Maybe you should check this value somewhere and raise an error if it is not `1`? That way, when a version `2` exists, you would provide a useful error message if someone tried to deserialize it, e.g. after a downgrade.

Reply:
Good point - I'll do this in `main` after the deserialize for the short term.  I've already made good headway into the code that will handle versions which will come as a fast-follow to this PR.

Reply:
I've moved this to the `from_path()` method here.  It will end up being in the versioning code to be out after this PR.